dns/zone.py

```diff
@@ -500,16 +501,27 @@ def to_file(self, f, sorted=True, relativize=True, nl=None):
 
         str_type = string_types
 
-        if nl is None:
-            opts = 'wb'
-        else:
-            opts = 'wb'
-
         if isinstance(f, str_type):
-            f = open(f, opts)
+            f = open(f, 'wb')
             want_close = True
         else:
             want_close = False
+
+        # must be in this way, f.encoding may contain None, or even attribute
+        # may not be there
+        file_enc = getattr(f, 'encoding', None)
+        if file_enc is None:
+            file_enc = 'utf-8'
```

Comment:
And when `f.encoding` is None (python 2, binary) or throws `AttributeError` we have binary mode, utf-8 may be default for text. This should be ascii as far as I understand.

On python2 with text mode, we have `None` too...

I suppose to just drop support for binary mode. In py2 we check for existence of `t` in `f.mode` and for py3 we check for an existing `f.encoding`: `if hasattr(f, 'encoding') or 't' in f.mode`.